packages/base/src/formbuilder/formselectors.ts

```diff
@@ -45,6 +52,7 @@ export function getSourceTypeForm(sourceType: SourceType): typeof BaseForm {
       break;
     // ADD MORE FORM TYPES HERE
   }
+  (SourceForm as any).sourceType = sourceType;
```

Comment:
Maybe we could add this property to `BaseForm`, that way we don't need to cast here. We can even pass the `sourceType` when we actually create the form

Reply:
Thanks!